### PR TITLE
operator mongodb-atlas-kubernetes (2.1.0)

### PR DESCRIPTION
**Pull request by josvazg**

### CODE REVIEW

File: operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+            ]
+          }
+        }
+      ]
+    capabilities: Full Lifecycle
+    categories: Database
+    description: The MongoDB Atlas Kubernetes Operator enables easy management of Clusters in MongoDB Atlas
+    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
+    operators.operatorframework.io/project_layout: go.kubebuilder.io/v2
+    containerImage: mongodb/mongodb-atlas-kubernetes-operator:2.1.0
```
Comment (josvazg) [inline operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml:130]:
```suggestion
    containerImage: quay.io/mongodb/mongodb-atlas-kubernetes-operator:2.1.0
```

File: operators/mongodb-atlas-kubernetes/2.1.0/metadata/annotations.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  operators.operatorframework.io.bundle.package.v1: mongodb-atlas-kubernetes
+  operators.operatorframework.io.bundle.channels.v1: stable
+  operators.operatorframework.io.bundle.channel.default.v1: stable
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.15.0+git
+  operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
+  operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v2
+
+  # Annotations for testing.
+  operators.operatorframework.io.test.mediatype.v1: scorecard+v1
+  operators.operatorframework.io.test.config.v1: tests/scorecard/
```
Comment (josvazg) [inline operators/mongodb-atlas-kubernetes/2.1.0/metadata/annotations.yaml:15]:
```suggestion
  operators.operatorframework.io.test.config.v1: tests/scorecard/
  com.redhat.openshift.versions: "v4.8"
```

File: operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,605 @@
...
+  labels:
+    operatorframework.io/arch.amd64: supported
+    operatorframework.io/arch.arm64: supported
+    operatorframework.io/os.linux: supported
+  name: mongodb-atlas-kubernetes.v2.1.0
+  namespace: placeholder
+spec:
+  apiservicedefinitions: {}
```
Comment (josvazg) [inline operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml:138]:
```suggestion
spec:
  relatedImages:
    - name: mongodb-atlas-kubernetes-operator-arm64
      image: quay.io/mongodb/mongodb-atlas-kubernetes-operator@sha256:42363c9c5ff727a92878a3c41721336344005ed98fcffbe9bfee8b34c8697cb2
    - name: mongodb-atlas-kubernetes-operator-amd64
      image: quay.io/mongodb/mongodb-atlas-kubernetes-operator@sha256:134eee3fcf0c41eac2ce7be1a891257714eba0b4b6ce54192f8985354f16cbb9
  apiservicedefinitions: {}
```

File: operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+                            fieldPath: metadata.name
+                      - name: OPERATOR_NAMESPACE
+                        valueFrom:
+                          fieldRef:
+                            fieldPath: metadata.namespace
+                      - name: WATCH_NAMESPACE
+                        valueFrom:
+                          fieldRef:
+                            fieldPath: metadata.annotations['olm.targetNamespaces']
+                    image: 134eee3fcf0c41eac2ce7be1a891257714eba0b4b6ce54192f8985354f16cbb9
```
Comment (josvazg) [inline operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml:523]:
```suggestion
                    image: quay.io/mongodb/mongodb-atlas-kubernetes-operator@sha256:134eee3fcf0c41eac2ce7be1a891257714eba0b4b6ce54192f8985354f16cbb9
```

**Comment by josvazg** [comment]
The `add-bundle-to-index` error is not even documented in the [troubleshooting guide](https://github.com/redhat-openshift-ecosystem/certification-releases/blob/main/4.9/ga/troubleshooting.md).

How do I fix it?

File: operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+      url: https://github.com/mongodb/mongodb-atlas-kubernetes
+  maintainers:
+    - email: support@mongodb.com
+      name: MongoDB, Inc
+  maturity: beta
+  provider:
+    name: MongoDB, Inc
+  version: 2.1.0
+  replaces: mongodb-atlas-kubernetes.v2.0.1
+  skipRange: ">=0.8.0"
```
Comment (slaskawi) [inline operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml:610]: This is the suspicious piece.

Comment (josvazg) [inline operators/mongodb-atlas-kubernetes/2.1.0/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml:610]:
```suggestion
```

**Comment by s-urbaniak** [comment]
Can someone help us here please? We get the following log output from the pipeline logs:
```
[add-bundle-to-index : add-bundle-to-index] + index --iib-url https://iib.engineering.redhat.com --indices registry.redhat.io/redhat/certified-operator-index:v4.16 --bundle-pullspec quay.io/operator-pipeline-prod/mongodb-atlas-kubernetes:2.1.0 --index-image-destination quay.io/operator-pipeline-prod/mongodb-atlas-kubernetes-index --image-output index-image-paths.txt --authfile /workspace/credentials/.dockerconfigjson --index-image-destination-tag-suffix=-df3a03d8ae4129b5e042fb2c4dfe27393bc8e18e
[add-bundle-to-index : add-bundle-to-index] 2024-02-20 12:50:25,616 [operator-cert] INFO Waiting for IIB batch build to finish: 655428
[add-bundle-to-index : add-bundle-to-index] 2024-02-20 12:50:45,718 [operator-cert] INFO Waiting for IIB batch build to finish: 655428
[add-bundle-to-index : add-bundle-to-index] 2024-02-20 12:51:05,825 [operator-cert] ERROR IIB build failed: 675027
[add-bundle-to-index : add-bundle-to-index] 2024-02-20 12:51:05,825 [operator-cert] INFO Reason: Failed to add the bundles to the index image: failed to add bundle "mongodb-atlas-kubernetes.v2.1.0": UNIQUE constraint failed: operatorbundle.name, operatorbundle.version, operatorbundle.bundlepath
[add-bundle-to-index : add-bundle-to-index] Traceback (most recent call last):
[add-bundle-to-index : add-bundle-to-index]   File "/home/user/.venv/bin/index", line 8, in <module>
[add-bundle-to-index : add-bundle-to-index]     sys.exit(main())
[add-bundle-to-index : add-bundle-to-index]              ^^^^^^
[add-bundle-to-index : add-bundle-to-index]   File "/home/user/.venv/lib/python3.11/site-packages/operatorcert/entrypoints/index.py", line 234, in main
[add-bundle-to-index : add-bundle-to-index]     iib_response = add_bundle_to_index(
[add-bundle-to-index : add-bundle-to-index]                    ^^^^^^^^^^^^^^^^^^^^
[add-bundle-to-index : add-bundle-to-index]   File "/home/user/.venv/lib/python3.11/site-packages/operatorcert/entrypoints/index.py", line 160, in add_bundle_to_index
[add-bundle-to-index : add-bundle-to-index]     raise RuntimeError("IIB build failed")
[add-bundle-to-index : add-bundle-to-index] RuntimeError: IIB build failed
```
What stands out is the "UNIQUE constraint" error which we don't understand.

**Comment by josvazg** [comment]
We noticed that, although PR #3203 was merged without issues, the certified image is still stuck at 2.0.0.
Maybe that is what breaks this PR as it was trying to replace 2.0.1, which might not be there?
Trying the hypothesis now.

**Comment by josvazg** [comment]
Confirmed: If we make 2.1 replace 2.0 instead of 2.0.1 (which was not properly ingested for some reason) then it works.

**Comment by josvazg** [comment]
Closing to let support unblock the situation

**Comment by s-urbaniak** [comment]
fyi @aksaswadkar we closed this PR to unblock the release pipeline for 2.0.1 as per https://github.com/redhat-openshift-ecosystem/certified-operators/pull/3203#issuecomment-1956368384

**Pull request merged**